torch/nn/modules/instancenorm.py

```diff
@@ -11,22 +11,36 @@ def __init__(self, num_features, eps=1e-5, momentum=0.1, affine=False,
     def _check_input_dim(self, input):
         return NotImplemented
 
-    def _load_state_dict_key_mismatch(self, full_name, name, is_missing):
-        if not is_missing and not self.track_running_stats and \
-                name in ('running_mean', 'running_var'):
-            raise KeyError(
-                'Unexpected running stats buffer "{name}" in state_dict for '
-                '{klass} with track_running_stats=False. If you are trying to '
-                'load a checkpoint saved before 0.4.0, this may be expected '
-                'because {klass} does not track running stats by default '
-                'anymore since 0.4.0. If running stats are not used, remove '
-                'them from state_dict before calling load_state_dict. '
-                'Otherwise, set track_running_stats=True in {klass} to use '
-                'running stats. See the documentation of {klass} for more '
-                'details.'
-                .format(name=full_name, klass=self.__class__.__name__))
-        super(_InstanceNorm, self)._load_state_dict_key_mismatch(
-            full_name, name, is_missing)
+    def _load_from_state_dict(self, state_dict, prefix, strict, missing_keys, unexpected_keys, error_msgs):
+        try:
+            version = state_dict._metadata[prefix]["version"]
```

Comment:
So we do use those weird keys now? Can we change them to `prefix[:-1]`? This will be an empty string for an empty string, and will remove the trailing dot otherwise.

Reply:
We have to use these weird keys, because you can do `m.add_module('', submodule)`

Reply:
What does that change? It will still slice out the last dot, I can't see the problem